src/Microsoft.AspNetCore.Mvc.Core/Routing/UrlHelper.cs

```diff
@@ -38,6 +39,17 @@ public UrlHelper(ActionContext actionContext)
             _routeValueDictionary = new RouteValueDictionary();
         }
 
+        public UrlHelper(ActionContext actionContext, ILinkGenerator linkGenerator)
```

Comment:
It might make sense for these to become separate implementations. 

For instance the `Router` property will throw an exception if called. I'm not sure how much common ground there is.

Reply:
+100 for different implementations

Reply:
Sure 👍 I too was leaning towards a separate UrlHelper initially.